src/main/java/org/mastodon/mamut/launcher/RecentProjectsPanel.java

```diff
@@ -128,24 +128,24 @@ private void remakeGUI( final Consumer< String > projectOpener )
 				gbc.gridy++;
 				gbc.gridx = 0;
 				final JTextArea ta = new JTextArea( projectPath );
-				ta.setEditable( false );
-				ta.setLineWrap( true );
-				ta.addMouseListener( new MouseDblClickOpenPath( projectPath ) );
-				add( ta, gbc );
+				ta.setEditable(true);
+				ta.setLineWrap(true);
+				ta.addMouseListener(new MouseDblClickOpenPath(ta.getText()));
+				add(ta, gbc);
 
 				gbc.gridx = 1;
 				final JButton btnOpen = new JButton( MastodonIcons.LOAD_ICON_SMALL );
 				btnOpen.addActionListener( l -> {
-					projectOpener.accept( projectPath );
+					projectOpener.accept(ta.getText());
 					// Recent projects will be updated in the launcher method.
 				} );
 				add( btnOpen, gbc );
 
 				gbc.gridx = 2;
 				final JButton btnClear = new JButton( MastodonIcons.REMOVE_ICON );
 				btnClear.addActionListener( l -> {
-					recentProjects.remove( projectPath );
-					remakeGUI( projectOpener );
+					recentProjects.remove(ta.getText());
```

Comment:
I guess $recentProjects.remove( projectPath )$ would be correct here. The resulting behavior would be more predictable. Could you please test this!

A text that was changed in the JTextArea can no longer be used to remove the correct entry from the recentProjects. Am I right?

Reply:
Yes, you are completely right. Thanks for spotting this. I changed it and also tested it.

Reply:
Ok, I tested it my self, and it now works. Perfect.